### ISSUE TITLE
Moveit Roarm M2 S

### ISSUE THREAD

**Issue by giwrgakhhs** [waveshareteam/roarm_ws_em0#5]
Good evening, I executed all the commands you have on github for the roarm m2 s the only difference is that I did dualboot and not in vurtial box. However when I execute the 4th step with moveit it shows me this error here. 
![1](https://github.com/user-attachments/assets/c9b75ade-08b4-4943-87ed-48d53e39aad4)
![2](https://github.com/user-attachments/assets/b539488e-99e1-4015-889d-989cf52b934c)
![3](https://github.com/user-attachments/assets/99d754a6-e7db-426e-957a-f9f98fdcf535)

**Comment by DUDULRX** [comment]
stop all the commands that are being executed.

1. Enter command 
```
top
```
to check whether any node in the background process is not stopped.
2. Enter command
 ```
ros2 topic list
```
to check whether interference exists in the same LAN

**Comment by giwrgakhhs** [comment]
![4](https://github.com/user-attachments/assets/4e9418bf-6b6c-4f4c-b01a-300043773183)
![5](https://github.com/user-attachments/assets/844d32df-ac37-450e-8279-deb31d2d35ae)

**Comment by DUDULRX** [comment]
And what was the order in which you executed the orders

**Comment by giwrgakhhs** [comment]
1 terminal 
sudo chmod 666 /dev/ttyUSB0
ros2 run roarm_driver roarm_driver
2 terminal 
ros2 launch roarm_moveit interact.launch.py

**Comment by DUDULRX** [comment]
you can execute it in this order
1 terminal
```
sudo chmod 666 /dev/ttyUSB0
```
```
ros2 run roarm_driver roarm_driver
```
2 terminal
```
ros2 launch roarm_description display.launch.py
```
Check whether the system reports an error

**Comment by giwrgakhhs** [comment]
Here it runs normally. (in the third step that is  "ros2 launch roarm_description display.launch.py" ) 
In the 4th step which is this command we have errrors "ros2 launch roarm_moveit interact.launch.py"

**Comment by DUDULRX** [comment]
Then you can take a complete screenshot of the terminal output, the terminal of the interact.launch

**Comment by giwrgakhhs** [comment]
[errors.txt](https://github.com/user-attachments/files/18151523/errors.txt)

**Comment by DUDULRX** [comment]
Check whether the is installed ros2 gripper controller
```
ros2 pkg list
```

**Comment by giwrgakhhs** [comment]
No, there doesn't seem to be such a package.

**Comment by DUDULRX** [comment]
You can try installing
```
sudo apt install ros-humble-gripper-controllers
```

**Comment by giwrgakhhs** [comment]
I have the same issue. I executed the order you sent me but it still won't run.
![7](https://github.com/user-attachments/assets/2f118ca5-2612-4218-97e0-47f3e5bb36df)

**Comment by DUDULRX** [comment]
Do you have Docker installed on your system? I have an image

**Comment by giwrgakhhs** [comment]
Yes, we have docker. Please forward us the image and commands we need to execute.

**Comment by DUDULRX** [comment]
1.Pull Image
```
docker pull dudulrx0601/ros_humble:roarm_m2

```
2.Creating a Container
```
docker run -it  --name roarm_m2 --privileged  --network host   -v=/dev:/dev   -v /tmp/.X11-unix:/tmp/.X11-unix  -e DISPLAY=unix$DISPLAY  dudulrx0601/ros_humble:roarm_m2
```
3.Set xhost on the host page and start the container
```
xhost +
``` 
```
docker start roarm_m2
```

4.Interactive Entry Container
```
docker exec -it roarm_m2 /bin/bash
```

**Comment by DUDULRX** [comment]
If you have any other questions, I'll answer them tomorrow

**Comment by giwrgakhhs** [comment]
With the docker it works fine, but it's very complicated and we are trying to use realsence camera to recognize objects and catch them so it's a bit strange that everything we want to do is done through the docker. Isn't there a way to solve it through normal linux software and not with images?

**Comment by DUDULRX** [comment]
Delete the previous one, pull the project again, install the dependencies, compile, and run

**Comment by giwrgakhhs** [comment]
Also, another question. Everything is working well and thank you for that. So how can I add files in this docker and correspondingly folders?

**Comment by DUDULRX** [comment]
You can install open-ssh in docker and enable root remote login, which should make uploading and editing files easier.
install
`
sudo apt install openssh-server
`

configuration
`
echo "Port 22">>/etc/ssh/sshd_config
`
`
echo "PermitRootLogin yes">>/etc/ssh/sshd_config
`

enable
`service ssh start
`

check
`service ssh status
`

or, you can use the docker cp command on the host to copy the file into the container.
`
docker cp test.py roarm_m2:/home/ws
`

**Comment by giwrgakhhs** [comment]
Good evening I saw on this link https://github.com/waveshareteam/roarm_ws that you added pick and place. We had an issue and you had sent us a docker to download it do you need to resend us an updated package that includes this feature or if you have one so we can work on it too?

**Comment by giwrgakhhs** [comment]
Good evening again, through the docker you sent us I made some changes by putting codes and acclimating realsense. It's for my diploma thesis which recognizes the red dice via camera and the bracket goes and catches them. However, since it's in a docker, I'd like to make it public so the next person who takes it on can handle the code. It's possible you could give me permission for the repository. Thank you very much.
